src/base_alloc/base_alloc_global.c

```diff
@@ -155,10 +155,19 @@ void *umf_ba_global_aligned_alloc(size_t size, size_t alignment) {
         return NULL;
     }
 
+    if (SIZE_MAX - size < ALLOC_METADATA_SIZE) {
```

Comment:
Maybe `(size > SIZE_MAX - ALLOC_METADATA_SIZE)` - it is more obvious

Reply:
Done